File: Programmers/HashMap/전화번호 목록.java
// 프로그래머스 전화번호 목록 해쉬

import java.util.*;
import java.io.*;
class Solution {
    public boolean solution(String[] phone_book) {
        HashSet<String> set = new HashSet<>();
        for (String str : phone_book) {
            set.add(str);
        }

        for (String phone : phone_book) {
            for (int i = 1; i < phone.length(); i++) {
                String prefix = phone.substring(0, i);
                if (set.contains(prefix)) {
                    return false;
                }
            }
        }

        return true;
    }
}

File: Programmers/HashMap/전화번호 목록.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <iostream>

using namespace std;

bool solution(vector<string> phone_book) {
    unordered_set<string> phone_set;

    for (auto phone : phone_book) {
        phone_set.insert(phone);
    }

    for (auto phone : phone_book) {
        for (int i = 0; i < phone.length(); i++) {
            string prefix = phone.substr(0, i);
            if (phone_set.count(prefix)) {
                return false;
            }
        }
    }
    return true;
}